hail/src/main/scala/is/hail/expr/ir/TypeCheck.scala

```diff
@@ -346,9 +346,11 @@ object TypeCheck {
       case CombOpValue(i, value, sig) => assert(value.typ == TBinary)
       case _: SerializeAggs =>
       case _: DeserializeAggs =>
-      case x@Begin(xs) =>
-        xs.foreach { x =>
-          assert(x.typ == TVoid)
+      case Begin(xs) =>
+        if (xs.length != 0) {
+          xs.dropRight(1).foreach { x =>
```

Comment:
Why do `dropRight` here when you do `init` everywhere else? is there a difference?